include/boost/histogram/algorithm/empty.hpp

```diff
@@ -0,0 +1,37 @@
+// Copyright 2019 Henry Schreiner
+//
+// Distributed under the Boost Software License, Version 1.0.
+// (See accompanying file LICENSE_1_0.txt
+// or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef BOOST_HISTOGRAM_ALGORITHM_EMPTY_HPP
+#define BOOST_HISTOGRAM_ALGORITHM_EMPTY_HPP
+
+#include <boost/histogram/fwd.hpp>
+#include <boost/histogram/indexed.hpp>
+
+namespace boost {
+namespace histogram {
+namespace algorithm {
+/** Check to see if all histogram cells are empty. Use coverage to include or
+  exclude the underflow/overflow bins.
+
+  All cells in the histogram are checked, and this returns as soon as one cell is
+  found that is not empty.
+
+  Returns true if all cells are empty, and false otherwise.
+ */
+template <class A, class S>
+auto empty(const histogram<A, S>& h, coverage cov) {
+  using value_type = typename histogram<A, S>::value_type;
+  value_type zero = value_type();
```

Comment:
This can be const and I think it should be "default_value" instead of "zero".

Reply:
Yes.